crates/database/src/orders.rs

```diff
@@ -401,13 +401,18 @@ pub fn user_orders<'a>(
     // queries are taking too long in practice.
     #[rustfmt::skip]
     const QUERY: &str = const_format::concatcp!(
-"SELECT ", ORDERS_SELECT,
+"(SELECT ", ORDERS_SELECT,
+" FROM ", ORDERS_FROM,
+" LEFT OUTER JOIN onchain_placed_orders onchain_o on onchain_o.uid = o.uid",
+" WHERE o.owner = $1) ",
```

Comment:
can't this be
```
 WHERE o.owner = $1 OR onchain_o.sender = $1
```
?

Reply:
test result with UNION: 
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 36 filtered out; finished in 310.63s

Test result without UNION:
test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 36 filtered out; finished in 328.50s

In the end, I went for your solution, as the test result difference is not big and might be due to other factors such as local usage.

Reply:
I looked at it in the query analyzer.  For some reason it doesn't use the right index without UNION. So let's keep it.